### PR TITLE
3.1 to 6.0 LTS track migration guide /3

### LINKED ISSUE #23817: Migration doc between LTS versions
It would be tremendously helpful if documentation was provided to migrate between LTS versions without having to migrate to each interim version (especially since all those workloads are optional/additional installs we otherwise don't need). For example, "Migrate from ASP.NET Core 3.1 to 6.0" in this case. Thanks for consideration.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 5bec61f9-7298-170d-5e12-abb27cde584e
* Version Independent ID: 27a43e3f-149d-9058-f937-0eda1d6dd4f2
* Content: [Migrate from ASP.NET Core 5.0 to 6.0](https://docs.microsoft.com/en-us/aspnet/core/migration/50-to-60?view=aspnetcore-6.0&tabs=visual-studio)
* Content Source: [aspnetcore/migration/50-to-60.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/migration/50-to-60.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-migration**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by Rick-Anderson**
Fixes #23817
[Internal review URL](https://review.docs.microsoft.com/en-us/aspnet/core/migration/31-to-60?view=aspnetcore-6.0&branch=pr-en-us-23854&tabs=visual-studio)
@guardrex  I'll merge this by 11/17 then you can add the Blazor stuff - or paste Blazor in an issue and I'll copy/paste.

Public review  of the doc build 
- download the [HTML files](https://www.dropbox.com/sh/yy994551cje0kap/AAB0tRnm7gze2kYrSYPSFHIda?dl=0)
- unzip
- view HTML file in a browser. It best to view the doc with the widest window that has no left or right TOC (Table  of contents)

### CODE REVIEW

**Comment by guardrex** [comment]
Recommend the announcement repo mention, too. I find that helpful in some of the app migration meltdown ☢️ situations. We have this nice bit in the migration topic ...

https://docs.microsoft.com/en-us/aspnet/core/migration/50-to-60?view=aspnetcore-5.0&tabs=visual-studio#review-breaking-changes

**Comment by Rick-Anderson** [comment]
@guardrex  I'm done pushing commit if you want to add Blazer info. Do me a favor and only have one Blazor H2 like you did in the 6.0 migration doc. The 3.1 has too many

image

**Comment by guardrex** [comment]
I added a section similar to what's in the 5.0 doc. We can't provide an all-feature update path. It's a crazy amount of churn 😵 to try and cover, as you can see by what happened there at 5.0. The recommendation is just to create a new 6.0 app, then migrate code and components using the *What's New* cross-links for specific guidance on individual features.

**Comment by Rick-Anderson** [comment]
@danroth27 can you assign some reviewers for this?

**Comment by Rick-Anderson** [comment]
@k3davis please review

**Comment by k3davis** [comment]
I haven't run it through a real-world scenario yet but I found this guide helpful, especially with respect to the DateTime binding change that I wasn't already aware of. I wasn't able to review the breaking changes list from the public download version.

I *really* appreciate the time spent putting this together, and so quickly. Thank you!

**Comment by Rick-Anderson** [comment]
> . I wasn't able to review the breaking changes list from the public download version.

You need to remove the `review.` from the URL. So it's https://docs.microsoft.com/en-us/dotnet/core/compatibility/5.0

File: aspnetcore/migration/31-to-60.md [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+
+You may need to delete the `bin` and `obj` folders. Run `dotnet nuget locals --clear all` to clear the NuGet package cache.
+
+## Minimal hosting model
+
+The ASP.NET Core templates generate code using the new [minimal hosting model](xref:migration/50-to-60#new-hosting-model). The minimal hosting model unifies *Startup.cs* and *Program.cs* into a single *Program.cs* file. `ConfigureServices` and `Configure` are no longer used. Apps migrating from ASP.NET Core 3.1 to 6.0 don't need to use the minimal hosting model, using `Startup` and the [Generic Host](xref:fundamentals/host/generic-host) used by the ASP.NET Core 3.1 templates is fully supported.
+
+To use `Startup` with the new minimal hosting model, see [Use Startup with the new minimal hosting model](xref:migration/50-to-60#smhm).
+
+To migrate to the new minimal hosting model using the following the pattern used by the ASP.NET Core 6.0 templates, see [
```
Comment (mkArtakMSFT) [inline aspnetcore/migration/31-to-60.md:88]:
```suggestion
To migrate to the new minimal hosting model using the following pattern used by the ASP.NET Core 6.0 templates, see [
```

**Review by mkArtakMSFT — APPROVED** [review]
Thanks @Rick-Anderson. Let's get this merged - I scanned through this and it looks fine to me. If issues discovered later on, we can address them then.

**Pull request merged**